simple-stack-example-kotlin-masterdetail-fragments/src/main/java/com/example/ktdagger/AndroidApp.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.ktdagger
+
+import android.content.Context
+import android.support.multidex.MultiDexApplication
+
+import io.realm.Realm
+
+class AndroidApp : MultiDexApplication() {
+
+    override fun onCreate() {
+        super.onCreate()
+        if (INSTANCE == null) {
+            INSTANCE = this
+            appComponent = DaggerAppComponent.builder()
+                .appModule(AppModule(applicationContext))
+                .build()
+        }
+        Realm.init(this)
+    }
+
+    companion object {
+        var appComponent: AppComponent? = null
+            private set
+        private var INSTANCE: AndroidApp? = null
+
+        operator fun get(context: Context): AndroidApp {
+            return context.applicationContext as AndroidApp
+        }
+
+        fun get(): AndroidApp? {
```

Comment:
This generally cannot be null when accessed anywhere else, so the var should be lateinit and this should be nonnull